internal/pkg/opa/default-compiler.go

```diff
@@ -93,84 +92,94 @@ func (compiler *policyCompiler) Configure(appConf *configs.AppConfig, compConf *
 
 // See Process() from opa.PolicyCompiler
 func (compiler policyCompiler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	// Extract uid from request
-	uid := util.GetRequestUID(req)
+	// Set start time for request duration
+	startTime := time.Now()
 
 	// Validate if policy compiler was configured correctly
 	if !compiler.configured {
-		compiler.handleError(w, uid, errors.Errorf("PolicyCompiler was not configured! Please call Configure(). "))
+		compiler.handleError(w, errors.Errorf("PolicyCompiler was not configured! Please call Configure(). "))
 		return
 	}
 
 	// Parse body of request
-	requestBody, bodyErr := compiler.parseRequestBody(uid, req)
+	requestBody, bodyErr := compiler.parseRequestBody(req)
 	if bodyErr != nil {
-		compiler.handleError(w, uid, bodyErr)
+		compiler.handleError(w, bodyErr)
 		return
 	}
 
 	// Extract input
 	for rootKey := range requestBody {
 		if rootKey != "input" {
-			logging.LogForComponent("policyCompiler").WithField("UID", uid).Warnf("Request field %q which will be ignored!", rootKey)
+			logging.LogForComponent("policyCompiler").Warnf("Request field %q which will be ignored!", rootKey)
 		}
 	}
 
 	rawInput, exists := requestBody["input"]
 	if !exists {
-		compiler.handleError(w, uid, internalErrors.InvalidInput{Msg: "PolicyCompiler: Incoming request had no field 'input'!"})
+		compiler.handleError(w, internalErrors.InvalidInput{Msg: "PolicyCompiler: Incoming request had no field 'input'!"})
 		return
 	}
 	input, ok := rawInput.(map[string]interface{})
 	if !ok {
-		compiler.handleError(w, uid, internalErrors.InvalidInput{Msg: "PolicyCompiler: Field 'input' in request body was no nested JSON object!"})
+		compiler.handleError(w, internalErrors.InvalidInput{Msg: "PolicyCompiler: Field 'input' in request body was no nested JSON object!"})
 		return
 	}
-	logging.LogForComponent("policyCompiler").WithField("UID", uid).Debugf("Received input: %+v", input)
+	logging.LogForComponent("policyCompiler").Debugf("Received input: %+v", input)
 
 	// Process path
 	output, err := compiler.processPath(input)
 	if err != nil {
-		compiler.handleError(w, uid, err)
+		compiler.handleError(w, err)
+		return
+	}
+
+	path, errPath := extractURLFromRequestBody(input)
+	method, errMethod := input["method"].(string)
```

Comment:
I would be careful with just accessing a map and casting the result into the type string. I would implement some additional method like extractURLFromRequestBody() which returns some useful error messages in case of a missing but required field (Because this field is passed by the user of our service it could be malformed in many interesting ways we should definitely be prepared for :D)